datafusion/expr/src/type_coercion/binary.rs

```diff
@@ -276,18 +280,16 @@ fn handle_timestamp_minus(lhs: &DataType, rhs: &DataType) -> Result<Option<DataT
         | (
             DataType::Timestamp(TimeUnit::Millisecond, _),
             DataType::Timestamp(TimeUnit::Millisecond, _),
-        ) => Ok(Some(DataType::Interval(IntervalUnit::DayTime))),
+        ) => Some(DataType::Interval(IntervalUnit::DayTime)),
         (
             DataType::Timestamp(TimeUnit::Microsecond, _),
             DataType::Timestamp(TimeUnit::Microsecond, _),
         )
         | (
             DataType::Timestamp(TimeUnit::Nanosecond, _),
             DataType::Timestamp(TimeUnit::Nanosecond, _),
-        ) => Ok(Some(DataType::Interval(IntervalUnit::MonthDayNano))),
-        (_, _) => Err(DataFusionError::Plan(
-            "The timestamps have different types".to_string(),
-        )),
+        ) => Some(DataType::Interval(IntervalUnit::MonthDayNano)),
+        (_, _) => None,
```

Comment:
To my understanding this is only "not supported yet", right? Or did I miss something as the function comment says this is an invalid case

Reply:
Yes, the subtraction of different timestamp types needs special handling. I think if the operation is done with second or millisecond-precision timestamps, `IntervalDayTime` type could be coerced, and for the other timestamp types `IntervalMonthDayNano` seems to be the only option.